src/Etg.Yams.Core/Process/Process.cs

```diff
@@ -65,6 +66,15 @@ public Process(AppIdentity identity, string exePath, bool showProcessWindow)
                 if (!_showProcessWindow)
                 {
                     _process.StartInfo.UseShellExecute = false;
+
+                    // If initialization installed something (such as nodejs) that modifies the machine path, the path wouldn't be found on the first run after imaging. 
+                    // Therefore we merge the current machine path into the process path.
+                    // This only works with UseShellExecute = false, UseShellExecute = true won't pass the process environment and creates a new one
+                    var processPath = Environment.GetEnvironmentVariable("PATH", EnvironmentVariableTarget.Process).TrimEnd(';');
+                    var machinePath = Environment.ExpandEnvironmentVariables(Environment.GetEnvironmentVariable("PATH", EnvironmentVariableTarget.Machine).TrimEnd(';'));
+                    var missing = machinePath.Split(';').Where(path => !string.IsNullOrEmpty(path)).Except(processPath.Split(';'));
+                    string mergedPath = missing.Aggregate(new StringBuilder(processPath), (result, next) => result.Append(';').Append(next)).Append(';').ToString();
+                    _process.StartInfo.EnvironmentVariables["PATH"] = mergedPath;
                 }
```

Comment:
@Maarten88 Thanks for contributing! This looks good but needs some cleanup as the code contains few duplication, a bit hard to read and is missing unit tests.

Here is what it could look like:
```csharp
                    // If initialization installed something (such as nodejs) that modifies the machine path, the path wouldn't be found on the first run after imaging. 
                    // Therefore we merge the current machine path into the process path.
                    // This only works with UseShellExecute = false, UseShellExecute = true won't pass the process environment and creates a new one
                    string processPath = GetPath(EnvironmentVariableTarget.Process);
                    string machinePath = GetPath(EnvironmentVariableTarget.Machine);
                    string mergedPath = MergePath(processPath, machinePath);
                    _process.StartInfo.EnvironmentVariables["PATH"] = mergedPath;
```

Private methods:
```csharp
        private string MergePath(string processPath, string machinePath)
        {
            IEnumerable<string> processPathSegments = SplitToSegments(processPath);
            IEnumerable<string> machinePathSegments = SplitToSegments(machinePath);
            IEnumerable<string> segmentsUnion = processPathSegments.Union(machinePathSegments);
            return string.Join(",", segmentsUnion);
        }

        private static IEnumerable<string> SplitToSegments(string processPath)
        {
            return processPath.Split(';').Where(path => !string.IsNullOrWhiteSpace(path));
        }

        private static string GetPath(EnvironmentVariableTarget target)
        {
            return Environment.GetEnvironmentVariable("PATH", target).TrimEnd(';');
        }
```

I'd recommend moving the path manipulation code to a different class so you can write some unit tests for it. You could also put `Environment.GetEnvironmentVariable` behind an interface so you can mock it and test the path merging code.

If that sounds like to much work, I appreciate the contribution anyway and I'll do the cleanup when I get a chance :) 

Also please feel free to add a note on `SetShowProcessWindow()` so others can benefit from what you found out and hopefully we can rename it on the next breaking change. Or mark it as `obsolete` and add a new method with a better name.

Reply:
That's much easier to understand than my version, only it is missing the finer details of the functionality, and is slower ;-). My code may be a bit unreadable indeed.

You're certainly right this needs tests, but the problem is that adding a few unittests will only prove the string manipulation, not the real problems that might arise: getting a too long large path on old windows versions, getting empty result when a variable is both present as machine and process var, running this on a linux one day with different environment sematics etc.

I'll try to split this out into a class and make it understandable. Maybe I'll even use string.Join ;-)